File: TelerikUIExamplesInCSharp/Examples/Calendar/LocalizedCalendar.cs
using System;

using Foundation;
using UIKit;

using TelerikUI;

namespace Examples
{
	public class LocalizedCalendar : ExampleViewController
	{
		public TKCalendar CalendarView {
			get;
			set;
		}
			
		public override void ViewDidLoad ()
		{
			this.AddOption ("Russian", SelectRussian);
			this.AddOption ("German", SelectGerman);
			this.AddOption ("Hebrew", SelectHebrew);
			this.AddOption ("Chinese", SelectChinese);
			this.AddOption ("Islamic", SelectIslamic);

			this.SelectedOption = 2;

			base.ViewDidLoad ();

			this.CalendarView = new TKCalendar (this.View.Bounds);
			this.CalendarView.AutoresizingMask = UIViewAutoresizing.FlexibleHeight | UIViewAutoresizing.FlexibleWidth;
			this.View.AddSubview (this.CalendarView);

			NSDate date = NSDate.Now;
			NSDateComponents components = new NSDateComponents ();
			components.Year = -1;
			this.CalendarView.MinDate = this.CalendarView.Calendar.DateByAddingComponents (components, date, NSCalendarOptions.None);
			components.Year = 1;
			this.CalendarView.MaxDate = this.CalendarView.Calendar.DateByAddingComponents (components, date, NSCalendarOptions.None);

			this.SelectHebrew (this, EventArgs.Empty);
		}

		public void SelectRussian (object sender, EventArgs e)
		{
			this.CalendarView.Calendar = new NSCalendar (NSCalendarType.Gregorian);
			this.CalendarView.Locale = new NSLocale ("ru_RU");
			this.CalendarView.ViewMode = TKCalendarViewMode.Month;
			((TKCalendarMonthPresenter)this.CalendarView.Presenter).Update (false);
		}

		public void SelectGerman (object sender, EventArgs e)
		{
			this.CalendarView.Calendar = new NSCalendar (NSCalendarType.Gregorian);
			this.CalendarView.Locale = new NSLocale ("de_DE");
			this.CalendarView.ViewMode = TKCalendarViewMode.Month;
			((TKCalendarMonthPresenter)this.CalendarView.Presenter).Update (false);
		}

		public void SelectHebrew (object sender, EventArgs e)
		{
			this.CalendarView.Calendar = new NSCalendar (NSCalendarType.Hebrew);
			this.CalendarView.Locale = new NSLocale ("he_IL");
			this.CalendarView.ViewMode = TKCalendarViewMode.Month;
			((TKCalendarMonthPresenter)this.CalendarView.Presenter).Update (false);
		}

		public void SelectChinese (object sender, EventArgs e)
		{
			this.CalendarView.Calendar = new NSCalendar (NSCalendarType.Chinese);
			this.CalendarView.Locale = new NSLocale ("zh_Hans_SG");
			this.CalendarView.ViewMode = TKCalendarViewMode.Month;
			((TKCalendarMonthPresenter)this.CalendarView.Presenter).Update (false);
		}

		public void SelectIslamic (object sender, EventArgs e)
		{
			this.CalendarView.Calendar = new NSCalendar (NSCalendarType.Islamic);
			this.CalendarView.Locale = new NSLocale ("ar-QA");
			this.CalendarView.ViewMode = TKCalendarViewMode.Month;
			((TKCalendarMonthPresenter)this.CalendarView.Presenter).Update (false);
		}
	}
}

File: TelerikUIExamplesInSwift2.0/TelerikUIExamplesInSwift/LocalizedCalendar.swift
//
//  LocalizedCalendar.swift
//  Swift Examples
//
//  Copyright (c) 2014 Telerik. All rights reserved.
//

import Foundation

class LocalizedCalendar: ExampleViewController {
    
    let calendarView = TKCalendar()
    
    override init(nibName nibNameOrNil: String?, bundle nibBundleOrNil: NSBundle?) {
        super.init(nibName: nibNameOrNil, bundle: nibBundleOrNil)

        self.addOption("Russian") { self.selectRussian() }
        self.addOption("German") { self.selectGerman() }
        self.addOption("Hebrew") { self.selectHebrew() }
        self.addOption("Chinese") { self.selectChinese() }
        self.addOption("Islamic") { self.selectIslamic() }
        
        self.selectedOption = 2
    }

    required init(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        self.calendarView.autoresizingMask = UIViewAutoresizing(rawValue:UIViewAutoresizing.FlexibleWidth.rawValue | UIViewAutoresizing.FlexibleHeight.rawValue)
        self.view.addSubview(self.calendarView)
        
        self.selectHebrew()
    }
    
    override func viewDidLayoutSubviews() {
        super.viewDidLayoutSubviews()
        self.calendarView.frame = self.exampleBounds
    }
    
    func selectRussian() {
        self.calendarView.calendar = NSCalendar(calendarIdentifier: NSGregorianCalendar)
        self.calendarView.locale = NSLocale(localeIdentifier: "ru_RU")
        self.calendarView.presenter().update(false)
    }
    
    func selectGerman() {
        self.calendarView.calendar = NSCalendar(calendarIdentifier: NSGregorianCalendar)
        self.calendarView.locale = NSLocale(localeIdentifier: "de_DE")
        self.calendarView.presenter().update(false)
    }
    
    func selectHebrew() {
        self.calendarView.calendar = NSCalendar(calendarIdentifier: NSHebrewCalendar)
        self.calendarView.locale = NSLocale(localeIdentifier: "he_IL")
        self.calendarView.presenter().update(false)
    }
    
    func selectChinese() {
        self.calendarView.calendar = NSCalendar(calendarIdentifier: NSChineseCalendar)
        self.calendarView.locale = NSLocale(localeIdentifier: "zh_Hans_SG")
        self.calendarView.presenter().update(false)
    }
    
    func selectIslamic() {
        self.calendarView.calendar = NSCalendar(calendarIdentifier: NSIslamicCalendar)
        self.calendarView.locale = NSLocale(localeIdentifier:"ar-QA")
        self.calendarView.presenter().update(false)
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
}
